lib/semantics/tools.h

```diff
@@ -111,6 +113,12 @@ inline bool IsOptional(const Symbol &symbol) {
 inline bool IsIntentIn(const Symbol &symbol) {
   return symbol.attrs().test(Attr::INTENT_IN);
 }
+inline bool IsIntentInOut(const Symbol &symbol) {
+  return symbol.attrs().test(Attr::INTENT_INOUT);
+}
+inline bool IsIntentOut(const Symbol &symbol) {
+  return symbol.attrs().test(Attr::INTENT_OUT);
+}
```

Comment:
Just curious.  What's the point of declaring functions to be ```inline```?

Reply:
https://en.cppreference.com/w/cpp/language/inline

Reply:
The compiler is free to ignore "inline" and to inline functions that do not specify it.  Given that, I see no value to specifying a function as ```inline```.

If there is some value, what criteria should I use to decide to declare a function ```inline```?

Reply:
A C++ compiler can't completely ignore `inline`; at the least, it works like `static` to prevent the name from becoming externally visible and causing link-time errors about multiple definitions.  Function definitions in header files outside class definitions need to be `static`, `inline`, or `constexpr`.